src/main/java/sg/ncl/MainController.java

```diff
@@ -1193,21 +1198,37 @@ public String checkJoinTeamInfo(
         userFields.put("id", session.getAttribute("id")); // ncl-id
         teamFields.put("name", teamPageJoinForm.getTeamName());
 
+        logger.info("Calling the registration service to do join team request");
         HttpHeaders headers = new HttpHeaders();
         headers.setContentType(MediaType.APPLICATION_JSON);
         headers.set("Authorization", AUTHORIZATION_HEADER);
 
         HttpEntity<String> request = new HttpEntity<>(mainObject.toString(), headers);
-        ResponseEntity responseEntity = restTemplate.exchange(properties.getSioRegUrl() + "/joinApplications", HttpMethod.POST, request, String.class);
-        
-        // perform join team request here
-        // add to user join team list
-        // ensure user is not already in the team or have submitted the application
-        // add to team join request map also for members approval function
-        User currentUser = userManager.getUserById(getSessionIdOfLoggedInUser(session));
-        int teamId = teamManager.getTeamIdByTeamName(teamPageJoinForm.getTeamName());
-        teamManager.addJoinRequestTeamMap2(getSessionIdOfLoggedInUser(session), teamId, currentUser);
-        return "redirect:/teams/join_application_submitted/" + teamId;
+        restTemplate.setErrorHandler(new MyResponseErrorHandler());
+        ResponseEntity response = restTemplate.exchange(properties.getSioRegUrl() + "/joinApplications", HttpMethod.POST, request, String.class);
+
+        String responseBody = response.getBody().toString();
+
+        try {
+            if (RestUtil.isError(response.getStatusCode())) {
+                MyErrorResource error = objectMapper.readValue(responseBody, MyErrorResource.class);
+
+                if (error.getName().equals(ExceptionState.TeamNotFoundException.toString())) {
+                    logger.warn("join team request : team name error");
+                    redirectAttributes.addFlashAttribute("message", "Team name does not exists.");
+                } else {
+                    logger.warn("join team request : some other failure");
+                    // possible sio or adapter connection fail
+                    redirectAttributes.addFlashAttribute("message", ERR_SERVER_OVERLOAD);
+                }
+                return "redirect:/teams/join_team";
+            }
+        } catch (IOException e) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)